Remove MIQ_SPARTAN in favor of Foreman

Fryguy opened the pull request:
MIQ_SPARTAN is no longer necessary now that we have Foreman and Procfiles, and in turn it complicates the worker count logic.  I propose we remove it.

cc @jrafanie @agrare

Fryguy commented:
Marked as WIP as I want to test it a little more.

Fryguy commented:
I'd like to get the UI team input on this as well... @h-kataria @himdel @skateman do you still use this?  If so, is the foreman way a sufficient alternative?

Fryguy commented on app/models/miq_schedule_worker.rb:
```diff
@@ -4,11 +4,7 @@ class MiqScheduleWorker < MiqWorker
-    return MiqServer.minimal_env_options.include?('schedule') ? 1 : 0 if MiqServer.minimal_env?
-    return 1
-  }
+  self.workers = 1
```
I'm not sure about this one...I think can remove it altogether, since you can't really change the worker count of miq_schedule?  @jrafanie Thoughts?

jrafanie replied:
Yeah, there's no harm of having the schedule running in whatever becomes of a spartan procfile

Fryguy replied:
@jrafanie, Ok, I can't remove it, technically, because then changing the value in settings will then cause MiqScheduleWorker.workers to return that value.  Even though the server role defines the max_concurrent as 1, I think that only means that only 1 server can have the _role_ concurrently, but within that 1 server if would run N workers.

Note that the other option here is that we can set `self.maximum_workers_count = 1`.  This value acts as a max clamp on the configured number, which would then allow 0 workers, which I don't think we want either.  The other reason I don't like that option is because the only reason it exists it to clamp the max number of web server workers per server (which now that I think about it sounds like a bad idea for podified :| ).

chessbyte commented:
When did I add MIQ_SPARTAN?  Happy that we are finally getting rid of it for better options.

Fryguy commented:
Funny, I thought I added it, and it turns out we were both wrong:

```
commit 9ebb9a8a4625a5e2fbca726d3072c0ce9d20ed3c
Author: Gregg Tanzillo <gtanzillo@manageiq.com>
Date:   Wed Apr 22 15:28:21 2009 +0000

    BugzID: 5439
    Added support for running in "spartan mode" where no workers are started. To enable spartan mode set the environment variable MIQ_SPARTAN=true

    git-svn-id: http://miq-ubuntusub.manageiq.com/svn/svnrepos/Manageiq/trunk@13770 3c68ef56-dcc3-11dc-9475-a42b84ecc76f
```

```
commit 6911bea0fc1f24fd6a3e2606a575feea820dd383
Author: Jason Frey <jfrey@redhat.com>
Date:   Wed Oct 28 15:57:56 2009 +0000

    Added ability to specify specific roles for MIQ_SPARTAN.

    git-svn-id: http://miq-ubuntusub.manageiq.com/svn/svnrepos/Manageiq/trunk@16861 3c68ef56-dcc3-11dc-9475-a42b84ecc76f
```

h-kataria commented:
> I'd like to get the UI team input on this as well... @h-kataria @himdel @skateman do you still use this? If so, is the foreman way a sufficient alternative?

@Fryguy I even forgot that it even existed, haven't used it in a long time.

jrafanie commented on app/models/miq_ui_worker.rb:
```diff
@@ -2,16 +2,6 @@ class MiqUiWorker < MiqWorker
-  self.workers = lambda do
-    if MiqServer.minimal_env?
-      # Force 1 UI worker in minimal mode, unless 'no_ui' is an option, which is
-      # done when the UI worker is debugged externally, such as in Netbeans.
```
🤣  Netbeans

Fryguy replied:
Cool bonus... this PR removes the last mention of Netbeans now 😆

himdel commented:
Agreed, it's mostly `simulate_queue_worker` for me, foreman works too, I haven't used `evm:start` for years.

skateman commented:
Never used this feature :thinking:

Fryguy commented:
Create documentation update PRs is the other repos.

Fryguy commented:
@jrafanie This is good to go.
